### PR TITLE
Investigate login error - Redirects to / by default

### PR DESCRIPTION
**Pull request by advayanand**
## Description

- Investigation:
  - Relevant file: api/src/controllers/users.ts 
  - Currently, redirection after logging in with Google is implemented by using the Referer header in the login request and redirecting to this URL. The Referer header stores the URL from which a login request was made: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referer
  - The Referer header being passed is undefined, causing a redirect to /undefined, which renders a 404 page. 
  - Another problem is that the req.session.returnTo field does not persist after a call to req.login(). This means that the field is undefined by the time successLogin is called, which causes a redirect to /undefined.
- Temporary Fix:
  - This PR redirects the user to / by default if the returnTo address is undefined, which is the case as mentioned above. 

## Steps to verify/test this change:
- Log in
- Verify that client is redirected back to / after login

## Final Checks:
- [ ] Verify successful deployment
- [ ] Delete branch

(optional)
- [ ] Write tests
- [ ] Write documentation

### CODE REVIEW

File: api/src/controllers/users.ts [hunk 1]
```diff
@@ -44,6 +46,7 @@ router.get('/auth/google',
  * Callback for Google authentication
  */
 router.get('/auth/google/callback', function (req, res) {
+  console.log(req.session);
```
Comment (ethanwong16) [inline api/src/controllers/users.ts:49]: If we're not using it could you comment this line out to minimize log sizes?

**Review by ethanwong16 — APPROVED** [review]
When you merge the PR, please keep the branch instead of deleting / squashing it so we can reference it in future advancements to the login.

Other than that LGTM!  Thanks @advayanand !

**Comment by ethanwong16** [comment]
Closes #243

File: api/src/controllers/users.ts (hunk 1 above)
Comment (advayanand) [reply api/src/controllers/users.ts]: Done

**Comment by ethanwong16** [comment]
5.5.2023 Discussion - @advayanand to merge hot fix since Ryan's Google auth changes are dependent on it + work with Jacob on improving login redirecting back to original page it was clicked from

**Pull request merged**